File: dagger/src/main/java/com/laioffer/lesson/使用手写dagger/MainActivity.java
package com.laioffer.lesson.使用手写dagger;

import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.laioffer.R;
import com.laioffer.lesson.使用手写dagger.模拟apt生成的代码.DaggerComputerComponent;
import com.laioffer.lesson.手写dagger.annotation.Inject;

/**
 * 类的描述: 注入目标
 */
public class MainActivity extends AppCompatActivity {

  @Inject // 第5个注解
  public Computer student;

  @Inject // 第5个注解
  public Computer student2;

  @Override
  protected void onCreate(Bundle savedInstanceState) { // 方法进栈
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);

    // 第4个注解@Component生成的类
    DaggerComputerComponent.create().inject(this);

    Toast.makeText(this, "student:" + student.hashCode(), Toast.LENGTH_SHORT).show();

    Toast.makeText(this, "student:" + student2.hashCode(), Toast.LENGTH_SHORT).show();
  }
}

File: fragment/fragment/src/main/java/com/laioffer/MainActivity.kt
package com.laioffer

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import com.laioffer.demo1.Demo1Activity
import com.laioffer.demo2.Demo2Activity
import com.laioffer.demo3.Demo3Activity
import com.laioffer.fragment.fragment.R

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    startNewActivity(R.id.demo1_activity, "Fragment基本使用", Demo1Activity::class.java)
    startNewActivity(R.id.demo2_activity, "动态添加Fragment", Demo2Activity::class.java)
    startNewActivity(R.id.demo3_activity, "管理Fragment", Demo3Activity::class.java)
  }

  private fun startNewActivity(btnId: Int, name: String, clazz: Class<*>) {
    val tv = findViewById<TextView>(btnId)
    tv.text = name
    tv.setOnClickListener {
      val intent = Intent(this@MainActivity, clazz)
      startActivity(intent)
    }
  }
}